src/transformState.js

```diff
@@ -6,6 +6,38 @@
  */
 function transformState(state, actions) {
   // write code here
-}
+  function addProps(add) {
+    for (const key in add) {
+      state[key] = add[key];
+    }
+  }
+
+  function removeProps(remove) {
+    for (const key of remove) {
+      delete state[key];
+    }
+  }
 
+  function clearAllProps() {
+    for (const key in state) {
+      delete state[key];
+    }
+  }
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can make a destructuring to get all needed props of action (type, extraData etc.)

Reply:
i am not sure if i got it right but you tell me :)